lib/Sources/App/SearchBarModel.swift

```diff
@@ -98,6 +107,16 @@ public final class SearchBarModel {
         let stripped = query.trimmingCharacters(in: .whitespacesAndNewlines)
         return stripped.count > 0 ? stripped : nil
     }
+
+    private func hostNameUrl(hostname: String) -> URL? {
+        guard (try? hostnameRegex?.wholeMatch(in: hostname)) != nil else {
+            return nil
+        }
+        guard gethostbyname(hostname) != nil else {
```

Comment:
What if the user is on a crappy internet connection and hostname lookup happens to fail at this moment (but might work on another attempt a short while later), what will happen in Topaz here?

Reply:
Their entry will be treated as a search query. But if their internet is that bad, who knows if they'd even be able to hit their search engine. I don't think there's much to be done about this? 🤷‍♂️

Reply:
I'm more just wondering if the user will hit a long delay before the app does anything if hostname lookup lags a bit?

Reply:
I think Chrome looks purely at the domain (similar to your regex) and doesn't involve hostname lookups at all, presumably to remove this complexity we've identified?

Background thread would prevent the app hang, but would we still show some kind of loading indicator to let them know something is actually happening?

Reply:
DNS resolution was something @davertay-j specifically asked for.

Yep! There's already a loading indicator that spins while this is all happening.

Reply:
The way Chrome does it is to just try to open it as a host if it looks like a host even if it isn't, and it fails with a DNS error. Is that better or worse than only opening it if it actually is a host? I have no idea, but that seemed like a reasonable approach.

If your internet is so bad that a hostname lookup hangs I doubt there is much benefit in adding more complexity to deal with that, but I suppose a simple thing to do is put an async timeout around it and just throw the string into the search engine as-per-usual if it takes more than a few seconds. The user should then get an error page anyway when the page fails to open probably.

Reply:
Will do.